### PR TITLE
Enhance/4761 - Implement modified snippet toggle component for enhanced AdSense setup flow

### PR DESCRIPTION
**Pull request by hussain-t**
## Summary

Addresses issue:

- #4761 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js
Comment (asvinb) [inline assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js:69]: @hussain-t Instead of duplicating the first part of the string, can we have it as a variable and then append the extra message to it?

File: assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.stories.js
Comment (asvinb) [inline assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.stories.js:42]: @hussain-t Since `SetupUseSnippetSwitch` does not accept any props, we can remove the `args` here.

**Review by asvinb — CHANGES_REQUESTED** [review]

File: assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js
Comment (hussain-t) [reply assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js:69]: IMO, it's better to keep it separated since it is more explicit and clean. Even if we append the extra message, we will have to translate the string.

File: assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js
Comment (hussain-t) [reply assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js:102]: Thanks, that's a good catch 👍

File: assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js
Comment (asvinb) [reply assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js]: Your comment makes sense, as you said, it's more clean and explicit. I'll let the merge reviewer take a look though :)

File: assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js
Comment (eugene-manuilov) [inline assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js:55]: I have some concerns that this hook will work correctly in "production" because the first call to the `getExistingTag` selector will always return `undefined` since the existing tag information is not "resolved" yet. It means that this hook will not run again once the existing tag information is collected and `existingTag` changes to either a string or `null` because `useEffectOnce` is an equivalent of the `useMount` hook.

What we need to do here is to ensure that we execute this hook when the existing tag is resolved.

File: assets/js/modules/adsense/datastore/settings.test.js
Comment (eugene-manuilov) [inline assets/js/modules/adsense/datastore/settings.test.js:363]: We can omit `status: 200` since this is the default value:
```suggestion
					{ body: response }
```

File: assets/js/modules/adsense/datastore/settings.test.js
Comment (eugene-manuilov) [inline assets/js/modules/adsense/datastore/settings.test.js:370]:
```suggestion
				expect( initialOriginalUseSnippet ).toBeUndefined();
```

File: assets/js/modules/adsense/datastore/settings.test.js
Comment (eugene-manuilov) [inline assets/js/modules/adsense/datastore/settings.test.js:388]:
```suggestion
				expect( originalUseSnippet ).toBe( response.useSnippet );
```

File: assets/js/modules/adsense/datastore/settings.test.js
Comment (eugene-manuilov) [inline assets/js/modules/adsense/datastore/settings.test.js:399]:
```suggestion
				).toBe( value );
```

File: assets/js/modules/adsense/datastore/settings.test.js
Comment (eugene-manuilov) [inline assets/js/modules/adsense/datastore/settings.test.js:419]:
```suggestion
				).toBe( value );
```

File: assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js
Comment (eugene-manuilov) [inline assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js:73]: There should be no whitespace at the end of translatable strings. Please, update it and the following translation.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @hussain-t. This looks almost good. Added some comments that we need to address before we can merge this ticket.

File: assets/js/modules/adsense/datastore/settings.test.js
Comment (hussain-t) [reply assets/js/modules/adsense/datastore/settings.test.js:363]: I think it makes more sense to have it explicitly in tests.

File: assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js
Comment (hussain-t) [reply assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js:73]: We need to append the extra string for `checkedMessage.` That's why I added the whitespace. Is it okay to add it at the beginning of a translatable string?

File: assets/js/modules/adsense/datastore/settings.test.js
Comment (eugene-manuilov) [reply assets/js/modules/adsense/datastore/settings.test.js:363]: Ok, let it be there then.

File: assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js
Comment (eugene-manuilov) [reply assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js:73]: There should be no extra spaces in a translatable string. If you need to add a space between two translated messages, just add it when you concat them: `message1 + ' ' + message2` or `${ message1 } ${ message2 }`

Comment (hussain-t) [reply assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js:73]: Alright, thanks!

File: assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js
Comment (eugene-manuilov) [inline assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js:50]: @hussain-t looks like we forgot one more thing here:

> The component should also include a useEffect that, if an existing tag was found, sets the useSnippet setting to false **and saves it right away** (unless of course that setting was already set to false before).

So, we need to save the settings if the current useSnippet setting is truthy and we change it to false.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]

File: assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js
Comment (hussain-t) [reply assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js:50]: Thanks, @eugene-manuilov, that's a good catch.

Comment (eugene-manuilov) [reply assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js:50]: @hussain-t we also need to save settings, which means that we need to call the `saveSettings` action: https://github.com/google/site-kit-wp/blob/4c156b533f57135c49dc64ff1adf2c1e143906cb/assets/js/modules/adsense/components/common/UseSnippetSwitch.js#L67

File: assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js
Comment (eugene-manuilov) [inline assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js:54]: Thanks, @hussain-t, but I think there is no need to wait till the saveSettings function call finishes. Let's remove await keyword and wrapping async function.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]

File: assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js
Comment (hussain-t) [reply assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js]: You're right 👍

**Review by eugene-manuilov — APPROVED** [review]
Ok, LGTM! Thanks, @hussain-t.

**Pull request merged**
